TeachingRecordSystem/src/TeachingRecordSystem.Core/Events/UserEditedEvent.cs

```diff
@@ -0,0 +1,7 @@
+namespace TeachingRecordSystem.Core.Events;
+
+public record UserEditedEvent : EventBase
+{
+    public required User User { get; init; }
+    public required Guid AddedByUserId { get; init; }
+}
```

Comment:
Need a `Changes` enum here too (like we use in ID) to track what actually changed